File: app/src/main/java/com/dinesh/recyclerview/java/nested/rv2/RvChildAdapter.java
package com.dinesh.recyclerview.java.nested.rv2;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.dinesh.recyclerview.R;

import java.util.List;

public class RvChildAdapter extends RecyclerView.Adapter<RvChildAdapter.ViewHolder> {

    List<String> items;

    public RvChildAdapter(List<String> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.nested_rv2_child_row, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.itemTextView.setText(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView itemTextView;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            itemTextView = itemView.findViewById(R.id.itemTextView);
        }
    }
}

File: app/src/main/java/com/dinesh/recyclerview/root/RvChildAdapter.kt
package com.dinesh.recyclerview.root

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.dinesh.recyclerview.R


class RvChildAdapter(var rvChildModelList: List<RvChildModel>, private val listener: RvChildInterface,var parentPosition: Int) : RecyclerView.Adapter<RvChildAdapter.ViewHolder>() {

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val TAG = "log_" + ViewHolder::class.java.name.split(ViewHolder::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]
        val rvChildCardView : CardView = itemView.findViewById(R.id.rvChildCardView)
        val tvChildTitle : TextView = itemView.findViewById(R.id.tvChildTitle)
        val tvChildDescription : TextView = itemView.findViewById(R.id.tvChildDescription)

        init {
            val itemClickListener = View.OnClickListener {
                it.context.startActivity(Intent(it.context, rvChildModelList[bindingAdapterPosition].className))
                listener.onChildItemClick(parentPosition,it, bindingAdapterPosition)
            }
            rvChildCardView.setOnClickListener(itemClickListener)
            tvChildTitle.setOnClickListener(itemClickListener)
            tvChildDescription.setOnClickListener(itemClickListener)


            val itemLongClickListener = View.OnLongClickListener {
                val isExpandable: Boolean = rvChildModelList[bindingAdapterPosition].isExpandable
                tvChildDescription.visibility = if (!isExpandable) View.VISIBLE else View.GONE
                rvChildModelList[bindingAdapterPosition].isExpandable = !rvChildModelList[bindingAdapterPosition].isExpandable
                true
            }

            rvChildCardView.setOnLongClickListener(itemLongClickListener)
            tvChildTitle.setOnLongClickListener(itemLongClickListener)
            tvChildDescription.setOnLongClickListener(itemLongClickListener)

        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val view: View = layoutInflater.inflate(R.layout.root_rv_child_row, parent, false)
        return ViewHolder(view)
    }


    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.tvChildTitle.text = rvChildModelList[position].title
        holder.tvChildDescription.text = rvChildModelList[position].description
    }

    override fun getItemCount(): Int {
        return rvChildModelList.size
    }
}
